Fix virtio-net-ccw

Jakob-Naucke opened the pull request:
The s390x architecture uses `virtio-net-ccw` instead of `virtio-net-pci`.

- Device number was formatted wrong in virtcontainers. Fix, and implement device structure like for PCI.
- Find device (by uevent or already existing) in agent, using the appropriate path for PCI or CCW.

This re-enables e.g.
```
$ docker run --runtime io.containerd.run.kata.v2 --rm busybox ping -c1 katacontainers.io
PING katacontainers.io (75.2.60.5): 56 data bytes
64 bytes from 75.2.60.5: seq=0 ttl=229 time=9.398 ms

--- katacontainers.io ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 9.398/9.398/9.398 ms
```
on s390x :)

cc @BbolroC @hbrueckner 

e: this definitely requires forward porting to runtime-rs, but from my first glance, the virtio-net implementation seems unfinished as of now.
e: sorry for the noise, lost some routine

hbrueckner commented on src/agent/src/device/network_device_handler.rs:
```diff
@@ -73,10 +84,53 @@ impl UeventMatcher for NetPciMatcher {
+            root_bus_path, device
+        );
+        NetCcwMatcher {
+            re: Regex::new(&re).expect("BUG: failed to compile NetCCWMatcher regex"),
```
nit: NetCcwMatcher (not ..CCW...)

hbrueckner reviewed (approved):
Thanks a lot. Just a small typo...

/lgtm

BbolroC reviewed (approved):
LGTM, thanks @Jakob-Naucke for your great work. I will take this into account for runtime-rs migration when the device hotplug is full fledged.

gkurz commented on src/runtime/virtcontainers/network.go:
```diff
@@ -271,7 +276,7 @@ func generateVCNetworkStructures(ctx context.Context, endpoints []Endpoint) ([]*
 			RawFlags:    noarp,
 			HwAddr:      endpoint.HardwareAddr(),
-			PciPath:     endpoint.PciPath().String(),
+			PciPath:     pciPath,
```
It is confusing to  reuse the PciPath field for non-PCI stuff. Can you please rename the field to something more abstract, e.g. DevicePath ?

gkurz reviewed (commented):
Hi @Jakob-Naucke !

Thanks for implementing proper network support for s390x. PR looks overall good to me, except the piggybacking of the PciPath interface to convey the CCW path to the agent. Please rename the field or add a new one.

BbolroC commented:
There are some checks failing both runtime (x86_64) and agent (s390x). Please get them to pass. 😉 

runtime
- https://github.com/kata-containers/kata-containers/actions/runs/13182368070/job/36796130072?pr=10817

agent
- https://github.com/kata-containers/kata-containers/actions/runs/13182368070/job/36796119844?pr=10817

Jakob-Naucke commented:
Thank you for your reviews! I've renamed `PciPath` in the protocol and fixed those failures specific to this PR (as far as I can tell) (sorry for force-push noise).

BbolroC commented:
> That's not entirely true. `run-cri-containerd (active, qemu)` fail across amd64 and s390x

I think it would be better to add a test for this change to the job. 😉

Jakob-Naucke commented:
The failures for k8s tests have (hopefully) been addressed – I had not accounted for situations where the device path was supposed to be empty.

> I think it would be good to add a test for this change to the job. 😉

Following internal discussion, I think it might be best to enable `tests/integration/docker` for s390x, which would incur a new workflow. Not sure this change would belong into this PR though.

BbolroC commented:
> Following internal discussion, I think it might be best to enable tests/integration/docker for s390x, which would incur a new workflow. Not sure this change would belong into this PR though.

Let's make the existing CIs green first. I think you could raise a follow-up PR for the workflow. But I am not sure what other reviewers think of this. @gkurz @hbrueckner

Jakob-Naucke commented:
Rebase without further edits.

BbolroC reviewed (approved):
LGTM thanks @Jakob-Naucke !

gkurz reviewed (approved):
/lgtm

Thanks @Jakob-Naucke !
